pkg/log/zap/flags.go

```diff
@@ -135,16 +142,51 @@ func (v levelValue) Type() string {
 	return "level"
 }
 
+func intLogLevel(l string) (int, error) {
+	lower := strings.ToLower(l)
+	var lvl int
+	switch lower {
+	case logLevelDebug:
+		lvl = -1
+	case logLevelInfo:
+		lvl = 0
+	case logLevelError:
+		lvl = 2
+	default:
+		i, err := strconv.Atoi(lower)
+		if err != nil {
+			return lvl, fmt.Errorf("invalid log level \"%s\"", l)
+		}
+
+		if i > 0 {
+			lvl = -1 * i
+		} else {
+			return lvl, fmt.Errorf("invalid log level \"%s\"", l)
+		}
+	}
+	return lvl, nil
+}
+
 type stackLevelValue struct {
 	set   bool
 	level zapcore.Level
 }
 
 func (v *stackLevelValue) Set(l string) error {
 	v.set = true
-	lvl, err := intLogLevel(l)
-	if err != nil {
-		return err
+	lower := strings.ToLower(l)
+	var lvl int
+	switch lower {
+	case logLevelDebug:
+		lvl = -1
+	case logLevelInfo:
+		lvl = 0
+	case logLevelError:
+		lvl = 2
+	case logLevelPanic:
+		lvl = 4
+	default:
+		return fmt.Errorf("invalid stacktrace level \"%s\"", l)
```

Comment:
With this change, we would no longer support integer-based stacktrace levels, which would be a regression of an existing feature. Can we still support integer stacktrace levels for values greater than 0?

Reply:
I put back previous behaviour, so supporting int log level in stack trace.